lib/HunkView.ml

```diff
@@ -31,43 +67,38 @@ let ui_hunk_summary (hunk : Patch.hunk) : Nottui.ui =
       at_symbols;
     ]
 
-let rec process_lines (mine_num : int) (their_num : int) (acc : Nottui.ui list)
-    = function
-  | [] -> List.rev acc
-  | line :: rest ->
-      let new_mine, new_their, ui_element =
-        match line with
-        | `Common line ->
-            let ui =
-              W.string ~attr:Notty.A.empty
-                (Printf.sprintf "%2d %2d   %s" (mine_num + 1) (their_num + 1)
-                   line)
-            in
-            (mine_num + 1, their_num + 1, ui)
-        | `Their line ->
-            let ui =
-              W.string
-                ~attr:Notty.A.(fg green)
-                (Printf.sprintf "   %2d + %s" (their_num + 1) line)
-            in
-            (mine_num, their_num + 1, ui)
-        | `Mine line ->
-            let ui =
-              W.string
-                ~attr:Notty.A.(fg red)
-                (Printf.sprintf "%2d    - %s" (mine_num + 1) line)
-            in
-            (mine_num + 1, their_num, ui)
-      in
-      process_lines new_mine new_their (ui_element :: acc) rest
+let process_lines hunk =
+  let mine, their = split_and_align_hunk hunk.Patch.lines in
+  let create_line_diff m t =
+    match (m, t) with
+    | Common m_text, Common t_text when m_text = t_text ->
+        WordDiff.CommonDiff m_text
+    | Common m_text, Common t_text ->
+        let diff = WordDiff.Diff.apply_word_diff m_text t_text in
+        WordDiff.ModifiedDiff (diff, diff)
```

Comment:
Isn't it weird that two aligned `Common` line are different ?
This cannot happen if `split_and_align_hunk` is correct and I think this impossible case could be eliminated if `split_and_align_hunk` was not used when computing word diffs.

Instead, collecting consecutive `Their` and `Mine` lines should be enough to compute word diffs. This would result in an API that is easier to understand and allows to reconstruct a value that looks a lot like the `hunk` type. My dream API for this would be:
```ocaml
val WordDiff.compute : Patch.hunk -> WordDiff.hunk
```
The `hunk` type is very easy to output, so a similar `WordDiff.hunk` type that contain the same information + the word diffs would be ideal to separate the code that computes the word diffs from the code that outputs them.